compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -2802,14 +2802,42 @@ abstract class GenJSCode extends plugins.PluginComponent
          *   {name1: arg1, name2: arg2, ... }
          */
 
+        @tailrec
+        def warnIfDuplicatedKey(pairs: List[(js.StringLiteral, js.Tree)],
+            namesCount: Map[String, Int] = Map.empty): Unit = {
+          pairs match {
+            case (js.StringLiteral(keyName), p) :: tail =>
+              val oldValue = namesCount.getOrElse(keyName, 0)
+              val newCount = namesCount.updated(keyName, oldValue + 1)
+              warnIfDuplicatedKey(tail, newCount)
+            case Nil =>
+              // Emmit all warnings
+              val duplicated = namesCount.filter(1 < _._2)
+              if(duplicated.nonEmpty) {
+                reporter.warning(pos,
+                  "Duplicate keys in object literal: " +
+                  duplicated.map{ case (keyName, count) =>
+                    s""""$keyName" defined $count times"""
+                  }.mkString(", ") +
+                  ". Only the last occurrence is assigned."
```

Comment:
I would still report one warning per key that is duplicate. So you get two warnings if you define `a` twice and `b` twice. (But you only get one warning if you define `a` 3 times.)